libr/config/config.c

```diff
@@ -83,39 +179,33 @@ R_API void r_config_list(RConfig *cfg, const char *str, int rad) {
 			cfg->cb_printf ("%s\n", node->name);
 		}
 		break;
+	case 'J':
+		verbose = true;
 	case 'j':
-		cfg->cb_printf ("{");
+		json = true;
+		json_is_first = true;
+		if (verbose) {
+			cfg->cb_printf ("[");
+		} else {
+			cfg->cb_printf ("{");
+		}
 		r_list_foreach (cfg->nodes, iter, node) {
 			if (!str || (str && (!strncmp (str, node->name, len)))) {
 				if (!str || !strncmp (str, node->name, len)) {
-					const char *val = node->value;
-					if (node->flags & CN_BOOL || node->flags & CN_INT || node->flags & CN_OFFT) {
-						if (!val) {
-							val = "0";
-						}
-						if (!strncmp (val, "0x", 2)) {
-							ut64 n = r_num_get (NULL, val);
-							cfg->cb_printf ("\"%s\":%"PFMT64d,
-								node->name, n);
-						} else if (r_str_isnumber (val) || !strcmp (val, "true") || !strcmp (val, "false")) {
-							cfg->cb_printf ("\"%s\":%s",
-								node->name, val);
-						} else {
-							char *sval = r_str_escape (val);
-							cfg->cb_printf ("\"%s\":\"%s\"", node->name, sval);
-							free (sval);
-						}
-					} else {
-						cfg->cb_printf ("\"%s\":\"%s\"",
-							node->name, val);
-					}
-					if (iter->n) {
+					if (!json_is_first) {
```

Comment:
dont use negated conditionals if possible. it reads better to focus on positive conditionals